File: Beginner/Lapindromes (LAPIN)/lapindromes.py
t = int(input())

while t:
    str = input()
    l = int(len(str))
    h = int(l/2)
    c = 0

    if l%2 == 0:
        a = str[:h]
        b = str[h:]
    
    else:
        a = str[:h]
        b = str[h+1:]

    a = list(a)
    b = list(b)

    for j in a :
        if j in b :
            b.remove(j)
        else :
            c = 1
    if c == 0 :
        print('YES')
    else :
        print('NO')

    t = t-1

File: Beginner/Lapindromes (LAPIN)/Lapindromes.c
#include <stdio.h>
#include <string.h>

int main(void) 
{
    // your code goes here
	int i,c,l,n,t,j;
    char s[1001];
    scanf("%d",&t);
    while(t--)
    {
        scanf("%s",s);
        l=strlen(s);
        if(l%2==0)
            n=l/2;
        else
            n=l/2+1;
        c=0;
        i=0;
        while(i<l/2)
        {
            j=n;
            while(j<l)
            {
                if(s[i]==s[j])
                {
                    c++;
                    s[j]=0;
                    break;
                }
                j++;
            }
            i++;
        }
        if(c==l/2)
            printf("YES\n");
        else
            printf("NO\n");
    }
	return 0;
}

